daemon panic on cosul restart

DEBU[0140] Watch triggered with 2 nodes                  discovery=consul
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0xad9361]

goroutine 127 [running]:
github.com/docker/libnetwork/datastore.func·001()
	/go/src/github.com/docker/docker/vendor/src/github.com/docker/libnetwork/datastore/datastore.go:267 +0x341
created by github.com/docker/libnetwork/datastore.(*datastore).Watch
	/go/src/github.com/docker/docker/vendor/src/github.com/docker/libnetwork/datastore/datastore.go:276 +0x4a9

--- a/datastore/datastore.go
+++ b/datastore/datastore.go
@@ -265,6 +265,13 @@ func (ds *datastore) Watch(kvObject KVObject, stopCh <-chan struct{}) (<-chan KV
 				close(sCh)
 				return
 			case kvPair := <-kvpCh:
+				// If the backend KV store gets reset libkv's go routine
+				// for the watch can exit resulting in a nil value in
+				// channel.
+				if kvPair == nil {
+					close(sCh)
+					return
+				}
 				dstO := ctor.New()
 
 				if err := dstO.SetValue(kvPair.Value); err != nil {

